tests/test_io_photogrammetry.py

```diff
@@ -0,0 +1,83 @@
+import os, tempfile
+import numpy as np
+import xarray as xr
+from collections import OrderedDict
+import cedalion
+import cedalion.dataclasses as cdc
+from cedalion.io import read_photogrammetry_einstar, read_einstar, opt_fid_to_xr
+from numpy.testing import assert_array_almost_equal
+
+
+def test_read_einstar():
+    # prepare test data
+    fiducials = np.random.rand(5, 3)
+    optodes = np.random.rand(np.random.randint(100), 3)
+    tmp_fn = write_test_photo_fn(fiducials, optodes)
+    # call read_einstar to read test data
+    fid, opt = cedalion.io.read_einstar(tmp_fn)
+    # evaluate
+    assert list(fid.keys()) == ["Nz", "Iz", "Rpa", "Lpa", "Cz"]
+    assert_array_almost_equal(np.array(list(fid.values())), fiducials)
+    assert_array_almost_equal(np.array(list(opt.values())), optodes)
+
+
+def test_opt_fid_to_xr():
+    for n_fidu in range(10):
+        for n_opts in range(10):
+            fiducials = np.random.rand(n_fidu, 3)
+            fiducials = {"F" + str(i): f for i, f in enumerate(fiducials)}
+            optodes = np.random.rand(n_opts, 3)
+
+            optodes = {"S" + str(i): o for i, o in enumerate(optodes)}
+            fid, opt = opt_fid_to_xr(fiducials, optodes)
+            assert isinstance(fid, xr.DataArray)
+            assert isinstance(opt, xr.DataArray)
+
+            if n_fidu > 0:
+                assert_array_almost_equal(fid.values, list(fiducials.values()))
+            else:
+                assert fid.shape == (0, 3)
+
+            if n_opts > 0:
+                assert_array_almost_equal(opt.values, list(optodes.values()))
+            else:
+                assert opt.shape == (0, 3)
```

Comment:
Previously, this test used randint to select the number of fiducials. `opt_fid_to_xr` did not expect to receive empty coordinate dicts and failed occasionally. When testing value ranges it is better to explicitly test each value.